compiler/semantic/op.go

```diff
@@ -410,19 +410,25 @@ func (a *analyzer) semOp(o ast.Op, seq dag.Seq) (dag.Seq, error) {
 	case *ast.From:
 		return a.semFrom(o, seq)
 	case *ast.Summarize:
-		keys, err := a.semAssignments(o.Keys, true)
+		keys, err := a.semAssignments(o.Keys)
 		if err != nil {
 			return nil, err
 		}
+		if assignmentHasDynamicLHS(keys) {
+			return nil, errors.New("summarize: keys must be static field references")
+		}
 		if len(keys) == 0 && len(o.Aggs) == 1 {
 			if seq := a.singletonAgg(o.Aggs[0], seq); seq != nil {
 				return seq, nil
 			}
 		}
-		aggs, err := a.semAssignments(o.Aggs, true)
+		aggs, err := a.semAssignments(o.Aggs)
 		if err != nil {
 			return nil, err
 		}
+		if assignmentHasDynamicLHS(keys) {
```

Comment:
These error cases should have test coverage.
```suggestion
		if assignmentHasDynamicLHS(aggs) {
```

Reply:
Indeed they should not in the pr.

Reply:
Hmm I guess I'll add them here since this wasn't actually possible before this pr